buildtest/system.py

```diff
@@ -23,20 +24,55 @@ def __init__(self):
         """Constructor method for BuildTestSystem(). Defines all system configuration using
            class variable **system** which is a dictionary
         """
+        self.logger = logging.getLogger(__name__)
+        self.logger.debug("Starting System Compatibility Check")
         self.init_system()
-        self.system["SYSTEM"] = platform.system()
+        self.system["platform"] = platform.system()
         self.scheduler = self.check_scheduler()
         self.check_lmod()
 
+        if self.system["platform"] != "Linux":
+            print("System must be Linux")
+            sys.exit(1)
+
+        self.logger.debug("Finished System Compatibility Check")
+
     def init_system(self):
         """Based on the module "distro" set the linux distrubution name and version
         """
-        self.system["OS_NAME"] = distro.linux_distribution(
+
+        self.system["os"] = {}
+
+        self.system["os"]["name"] = distro.linux_distribution(
             full_distribution_name=False
         )[0]
-        self.system["OS_VERSION"] = distro.linux_distribution(
+        self.system["os"]["version"] = distro.linux_distribution(
             full_distribution_name=False
         )[1]
+        self.system["os"]["major_version"] = distro.major_version(best=True)
+        self.system["os"]["minor_version"] = distro.major_version(best=True)
+        self.system["os"]["build_number"] = distro.build_number(best=True)
+        self.system["env"] = dict(os.environ)
+        self.system["python"] = shutil.which("python") or None
```

Comment:
I think if shutil.which() can't find something it already returns None.

```python
import shutil                                                           
res = shutil.which('nope')                                              
res == None                                                             
True
```
so the "or None" is not necessary.

Reply:
Also now we need to be very careful for what is done with:
```python
self.system["env"] = dict(os.environ)
```
because what goes in the environment? Secrets, of course.

Reply:
i fixed the shutil issue. 

Oh yea i know if i figure capturing environment shell can be useful for debugging for both end-users and maintainers